posts/inside-rust/2023-11-09-spec-vision.md

```diff
@@ -0,0 +1,206 @@
+Such bounds allow for a middle ground of programs, where we do not commit all future versions of Rust to always make the same decision that the current version makes.
+For example, one can then say, prescriptively, that a given grammar provides a lower bound on the set of programs that must be accepted by all future versions of Rust, while still allowing the language to evolve in a backward compatible fashion.
+One can also say, descriptively, that the current version of Rust rejects source inputs that do not conform to the grammar.
+
+The descriptive definitions tell the reader how a construct will behave with respect to that Rust version; the prescriptive bounds tell the reader what they can and cannot expect to hold true in the future of Rust.
+
+Thus, we conclude that an ideal specification will need to address both the static and dynamic semantics, for both the prescriptive bounds and descriptive details.
+
+**"delegates"**: There are broad areas where the questions of what semantics we want, and how they should be specified, are open research topics.
+Examples of such areas include: macros 2.0, the type inference rules, the trait matching rules, and the operational semantics of unsafe code.
+It is not reasonable for the specification team to claim authority on such topics.
+Instead, other teams will be invited to contribute their own detailed descriptions, which can be published as their own documents that the specification can reference.
+Each such document is, like the specification itself, coupled to a specific Rust version.
+Furthermore, each such document is analogous to the detailed descriptions: the scope of each document produced via delegation is intended to be restricted to a specific Rust version.
+
+If a contributing team has input about broader prescriptive rules that should hold beyond the current Rust version, then that should be be part of the Rust specification document itself.
+Such prescriptive rules should always be the responsibility of the specification team to incorporate into the document.
+All such prescriptive rules are then subject to the specification approval process.
+
+</blockquote>
+</details>
+
+<p></p>
+
+## Incremental Development
+
+It is ambitious to provide both prescriptive bounds for current and future Rust versions and descriptive details of the current Rust version.
+We will maximize the value of our efforts by working iteratively and incrementally.
+The specification can have gaps where the prescribed bounds are broader than necessary.
+Subsequent releases of the specification can tighten those prescribed bounds.
+
+We expect early versions of the spec to focus heavily on delivering the detailed description of the current Rust version.
+Such a specification could be derived heavily from an existing work product, such the Ferrocene specification, since that explicitly focuses on providing a detailed description of a specific Rust version. 
```

Comment:
```suggestion
Such a specification could be derived heavily from an existing work product, such as the Ferrocene specification, since that explicitly focuses on providing a detailed description of a specific Rust version. 
```